main/version_checker.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/fatih/color"
+	log "github.com/sirupsen/logrus"
+)
+
+const currentVersion = "v0.1.4"
+
+type VersionChecker struct {
+	// httpClient is a http client shared across all http requests within this package.
+	httpClient http.Client
+}
+
+func (v *VersionChecker) checkCloudConciergeVersion() {
+	latestVersion, err := v.getLatestCloudConciergeVersion()
+	if err != nil {
+		log.Errorf("[version_checker] Error checking latest cloud concierge version: %s", err.Error())
+		return
+	}
+
+	cloudConciergeTitle := fmt.Sprintf("################################################################# Cloud Concierge %s ##################################################################", currentVersion)
+	footer := "###########################################################################################################################################################"
+
+	if latestVersion != currentVersion {
+		color.Red(cloudConciergeTitle)
+		color.Red(fmt.Sprintf("# You are currently running version %s of cloud-concierge, the latest version is %s, run docker pull dragondrop/cloud-concierge:latest to update. #", currentVersion, latestVersion))
+		color.Red(footer)
+	} else {
+		color.Green(cloudConciergeTitle)
+		color.Green(fmt.Sprintf("############################### You are currently running version %s of cloud-concierge, this is the latest version. ##################################", currentVersion))
+		color.Green(footer)
+	}
+}
+
+type GithubTagResponse struct {
+	TagName string `json:"tag_name"`
+}
+
+func (v *VersionChecker) getLatestCloudConciergeVersion() (string, error) {
+	log.Debugf("Checking for latest version of cloud-concierge...")
+	response, err := v.httpClient.Get("https://api.github.com/repos/dragondrop-cloud/cloud-concierge/releases/latest")
```

Comment:
Does this handle the following edge case:
- We push a new, pre-prod release (say v0.1.5-beta).

We basically want to guarantee that that the user is only recommended to upgrade to a production-ready release, not a beta release.

May need to remove the `/latest` suffix here and then iterate for the first tag that does not have a `-beta` suffix.